pkg/utils/utils.go

```diff
@@ -515,3 +519,95 @@ func CompareStringAndByteSlices(valueString *string, valueByte []byte) bool {
 
 	return bytes.Equal(valueByte, []byte(*valueString))
 }
+
+// CreateCertOpts contains options for a cert pool creation.
+type CreateCertOpts struct {
+	CABundle   []byte
+	CAProvider *esv1beta1.CAProvider
+	StoreKind  string
+	Namespace  string
+	Client     client.Client
+}
+
+// FetchCACertFromSource creates a CertPool using either a CABundle directly, or
+// a ConfigMap / Secret.
+func FetchCACertFromSource(ctx context.Context, opts CreateCertOpts) ([]byte, error) {
+	if len(opts.CABundle) == 0 && opts.CAProvider == nil {
+		return nil, nil
+	}
+
+	if len(opts.CABundle) > 0 {
+		pem, err := base64decode(opts.CABundle)
```

Comment:
Is this why the test Gustavo mentioned needed to change?

Reply:
Most likely, yes. It's strange, because most of the code was decoding it, so I think I was doing this on purpose, but I have to verify. And if yes, this might indeed be a breaking change then.

Reply:
Actually. I'm just gonna AUTO decode this thing. :D :D :D If it's base64 decode it, if not, leave it as is. DONE.

Reply:
Added some further checks to make sure that the data is a valid certificate.